### PR TITLE
SoundWire: ADR quirks take 2

### PR DESCRIPTION
**Pull request by plbossart**
second version to address upstream comments.
Replaces #2743

### CODE REVIEW

**Review by RanderWang — APPROVED** [review]

File: drivers/soundwire/slave.c [hunk 1]
```diff
@@ -169,23 +105,14 @@ static bool find_slave(struct sdw_bus *bus,
 	if (ACPI_FAILURE(status)) {
 		dev_err(bus->dev, "_ADR resolution failed: %x\n",
 			status);
-		return false;
+		return 0;
```
Comment (bardliao) [inline drivers/soundwire/slave.c]: `bool find_slave()` should return false.

**Review by bardliao — CHANGES_REQUESTED** [review]
`bool find_slave()` should return false instead of 0.

File: drivers/soundwire/slave.c (hunk 1 above)
Comment (plbossart) [reply drivers/soundwire/slave.c]: ah yes, this was changed to long long but I changed it back to a bool without changing the return value. Will fix.

Comment (plbossart) [reply drivers/soundwire/slave.c]: fixed now

**Review by bardliao — APPROVED** [review]

**Review by RanderWang — APPROVED** [review]

**Review by lyakh — APPROVED** [review]

**Pull request merged**